core/src/com/unciv/ui/worldscreen/unit/UnitActions.kt

```diff
@@ -12,382 +12,354 @@ import com.unciv.models.UncivSound
+                                tile.turnsToImprovement = 2
+                            }
+                            tile.improvement = null
+                            if (!unit.hasUnique("No movement cost to pillage")) unit.useMovementPoints(1f)
+                            unit.healBy(25)
                         }
-                        unit.destroy()
-                    })
-        }
 
-        if (unit.name == "Great Merchant" && !unit.isEmbarked()) {
-            val canConductTradeMission = tile.owningCity?.civInfo?.isCityState() == true
-                    && tile.owningCity?.civInfo?.isAtWarWith(unit.civInfo) == false
-                    && unit.currentMovement > 0
-            actionList += UnitAction(
-                    type = UnitActionType.ConductTradeMission,
-                    canAct = canConductTradeMission,
-                    uncivSound = UncivSound.Chimes,
-                    action = {
-                        // http://civilization.wikia.com/wiki/Great_Merchant_(Civ5)
-                        var goldEarned = (350 + 50 * unit.civInfo.getEra().ordinal) * unit.civInfo.gameInfo.gameParameters.gameSpeed.modifier
-                        if (unit.civInfo.policies.isAdopted("Commerce Complete"))
-                            goldEarned *= 2
-                        unit.civInfo.gold += goldEarned.toInt()
-                        val relevantUnique = unit.getUniques().first { it.startsWith("Can undertake") }
-                        val influenceEarned = Regex("\\d+").find(relevantUnique)!!.value.toInt()
-                        tile.owningCity!!.civInfo.getDiplomacyManager(unit.civInfo).influence += influenceEarned
-                        unit.civInfo.addNotification("Your trade mission to [${tile.owningCity!!.civInfo}] has earned you [${goldEarned.toInt()}] gold and [$influenceEarned] influence!", null, Color.GOLD)
-                        unit.destroy()
-                    })
-        }
-    }
-
-    private fun addFortifyActions(actionList: ArrayList<UnitAction>, unit: MapUnit, unitTable: UnitTable) {
-
-        val action = UnitAction(
-                type = UnitActionType.Fortify,
-                canAct = unit.currentMovement > 0,
-                uncivSound = UncivSound.Fortify,
-                action = {
-                    unit.fortify()
-                    unitTable.selectedUnit = null
-                })
-
-        if (unit.health < 100) {
-            val actionForWounded = action.copy(
-                    type = UnitActionType.FortifyUntilHealed,
-                    title = UnitActionType.FortifyUntilHealed.value,
-                    action = {
-                        unit.fortifyUntilHealed()
-                        unitTable.selectedUnit = null
-                    })
-            actionList += actionForWounded
-        }
+    private fun getLambdaOrNull(test: Boolean, lambda: () -> Unit):(() -> Unit)? =
+            if (test) lambda else null
```

Comment:
had to make this function to wrap lambdas or the type inference would fail, unless you know of a better way?